src/test/java/GmailServletTest.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.api.services.gmail.model.Message;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.model.AuthenticationVerifier;
+import com.google.sps.model.GmailClient;
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.servlets.GmailServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/**
+ * Test Gmail Servlet to ensure response contains correctly parsed messageIds. Assumes
+ * AuthenticatedHttpServlet is functioning properly (those tests will fail otherwise).
+ */
+@RunWith(JUnit4.class)
+public final class GmailServletTest {
+  private GmailClient gmailClient;
+  private GmailServlet servlet;
```

Comment:
Group instance and static variables together

Reply:
Agreed - fixed!